### PR TITLE
fix: add field from scale-range to aggregate groupby

### LINKED ISSUE #7486: field as a scale range is not added to group by of aggregation
Docs (https://vega.github.io/vega-lite/docs/scale.html#range) say that color values can be taken directly from data:
> For discrete and discretizing scales, an array of desired output values or an object with a field property representing the range values. For example, if a field color contains CSS color names, we can set range to {field: "color"}.

When I try this, nothing is shown in the plot: [example](https://vega.github.io/editor/#/url/vega-lite/N4IgJgpgzgxgTgSwA4BcEHsB2IBcIDKK6MA1gARKIzRnoBmZArGQCoQwAWZAwugLZIAhpgQ10ANwhxWCPhAB0IADThBKQblABXOABtcq9QHooRUlHkwo4kAF8VKOMKh10cPrgDaoXenQktJAMoAE8+ACN0fRU6OH5NQw0cUHFBXS1oL1BQiKiDAEF8gAUAGWUQGCi3AzgIMDslbLDI-Tx8gFkALQA5csrfOANw3UFSBqbc1pB2-AAxFj6qwbwAc1qIbHsJloMASQAhdsWBofSIcZAcnbwAcQB5O5vj6rwkHSRdc9sAXXsQEggIWCzTyMVEujAUC8FSWIF+PxUfEEcBICRQISQ5zwugQmHOKiQ6FxKFwjgyfw2lTAuJWCQAHmjZBAAKoiEl4EIQZHlOjg+p4MBqL4qIHJECCFZrCArIUGOTCHl8gyUBDUcrozEGACOWmEaHUaEkDRhJzFvIgEOBk3VGKxIEw-FxaXKsDSWNATkwK3dIHNlrw-WqtmDwaAA). Data viewer suggests that the `color` field doesn't make it into the resulting dataset.

If I bind this field (`color`) to another property, everything suddenly works fine! [Example](https://vega.github.io/editor/#/url/vega-lite/N4IgJgpgzgxgTgSwA4BcEHsB2IBcIDKK6MA1gARKIzRnoBmZArGQCoQwAWZAwugLZIAhpgQ10ANwhxWCPhAB0IADThBKQblABXOABtcq9QHooRUlHkwo4kAF8VKOMKh10cPrgDaoXenQktJAMoAE8+ACN0fRU6OH5NQw0cUHFBXS1oL1BQiKiDAEF8gAUAGWUQGCi3AzgIMDslbLDI-Tx8gFkALQA5csrfOANw3UFSBqbc1pB2-AAxFj6qwbwAc1qIbHsJloMASQAhdsWBofSIcZAcnbwAcQB5O5vj6rwkHSRdc9sAXXsQEggIWCzTyMVEujAUC8FSWIF+PxUfEEcBICRQISQ5zwugQmHOKiQ6FxKFwjgyfw2lTAuJWCQAHmjZBAAKoiEl4EIQZHlOjg+p4MBqL4qIHJECCFZrCArIUGOTCHl8gyUBDUcrozEGACOWmEaHUaEkDRhJzFvIgEOBk3VGKxIEw-FxaXKsDSWNATkwK3dIHNlrw-WqtmDKnQQhgCHRCT9-JN1RUrs+CU93q8AEYlGnfipPt7MPzMFpdLpg7YgA).

As I undestand, both examples linked above are supposed to give the same result.

### PR DESCRIPTION
**Pull request by donghunc**
fixes #7486

### CODE REVIEW

File: src/compile/compile.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import {
 import {Dict, keys} from '../util';
 import {buildModel} from './buildmodel';
 import {assembleRootData} from './data/assemble';
+//import {drawDataflow} from './data/debug';
```
Comment (domoritz) [inline src/compile/compile.ts:21]: These unrelated changes should not be in the pull request.

**Review by domoritz — CHANGES_REQUESTED** [review]
Please include the changes from https://github.com/vega/vega-lite/pull/7526

File: src/compile/compile.ts [hunk 2]
```diff
@@ -111,12 +111,12 @@ export function compile(inputSpec: TopLevelSpec, opt: CompileOptions = {}) {
     // Please see inside model.parse() for order of different components parsed.
     model.parse();
 
-    // drawDataflow(model.component.data.sources);
```
Comment (domoritz) [inline src/compile/compile.ts:114]: Remove this from the diff

**Review by domoritz — CHANGES_REQUESTED** [review]

File: src/compile/compile.ts (hunk 2 above)
Comment (donghunc) [reply src/compile/compile.ts]: Resolved.

File: src/compile/compile.ts (hunk 1 above)
Comment (donghunc) [reply src/compile/compile.ts]: Resolved.

**Review by domoritz — CHANGES_REQUESTED** [review]
This is still missing the change from https://github.com/vega/vega-lite/pull/7526/files#diff-245098c24d290d1cd456956b3a74d688f56ec8a89e28ca9851ddd21067d8518aR525

**Review by domoritz — APPROVED** [review]

**Pull request merged**